kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -68,6 +76,61 @@ def contracts(self) -> Dict[str, Contract]:
             _contracts[contract_name] = Contract(contract_name, contract_json, foundry=True)
         return _contracts
 
+    @cached_property
+    def contract_ids(self) -> Dict[int, str]:
+        _contract_ids = {}
+        for c in self.contracts.values():
+            _contract_ids[c.contract_id] = c.name
+        return _contract_ids
+
+    def srcmap_data(self, contract_name: str, pc: int) -> Optional[Tuple[Path, int, int]]:
+        if contract_name not in self.contracts:
+            _LOGGER.info(f'Contract not found in Foundry project: {contract_name}')
+        contract = self.contracts[contract_name]
+        if pc not in contract.srcmap:
+            _LOGGER.info(f'pc not found in srcmap for contract {contract_name}: {pc}')
+            return None
+        s, l, f, _, _ = contract.srcmap[pc]
+        if f not in self.contract_ids:
+            _LOGGER.info(f'Contract id not found in sourcemap data: {f}')
+            return None
+        src_contract = self.contracts[self.contract_ids[f]]
+        # TODO: Hardcoding that out directory is just one deep in project
```

Comment:
This is an indication that we should be taking the root of the foundry directory as input instead (or assuming `.`), and reading the appropriate `out` directory from the `foundry.toml`.

Then the directory to run in can become an optional argument to all our `foundry` commands.

Reply:
Issue created: https://github.com/runtimeverification/evm-semantics/issues/1630